pkg/authorization/apibinding_authorizer.go

```diff
@@ -98,33 +105,68 @@ func (a *apiBindingAccessAuthorizer) Authorize(ctx context.Context, attr authori
 	// get the cluster from the ctx.
 	lcluster, err := genericapirequest.ClusterNameFrom(ctx)
 	if err != nil {
+		kaudit.AddAuditAnnotations(
+			ctx,
+			APIBindingContentAuditDecision, DecisionNoOpinion,
+			APIBindingContentAuditReason, fmt.Sprintf("error getting cluster from request: %v", err),
+		)
 		return authorizer.DecisionNoOpinion, apiBindingAccessDenied, err
 	}
 
 	bindingLogicalCluster, bound, err := a.getAPIBindingReference(attr, lcluster)
 	if err != nil {
+		kaudit.AddAuditAnnotations(
+			ctx,
+			APIBindingContentAuditDecision, DecisionNoOpinion,
+			APIBindingContentAuditReason, fmt.Sprintf("error getting API binding reference: %v", err),
+		)
 		return authorizer.DecisionNoOpinion, apiBindingAccessDenied, err
 	}
 
 	if !bound {
+		kaudit.AddAuditAnnotations(
+			ctx,
+			APIBindingContentAuditDecision, DecisionAllowed,
+			APIBindingContentAuditReason, "no API binding found",
+		)
 		return a.delegate.Authorize(ctx, attr)
 	}
 
 	apiExport, found, err := a.getAPIExport(bindingLogicalCluster)
 	if err != nil {
+		kaudit.AddAuditAnnotations(
+			ctx,
+			APIBindingContentAuditDecision, DecisionNoOpinion,
+			APIBindingContentAuditReason, fmt.Sprintf("error getting API export: %v", err),
+		)
 		return authorizer.DecisionNoOpinion, apiBindingAccessDenied, err
 	}
 
 	// If we can't find the export default to close
 	if !found {
+		kaudit.AddAuditAnnotations(
+			ctx,
+			APIBindingContentAuditDecision, DecisionNoOpinion,
+			APIBindingContentAuditReason, fmt.Sprintf("API export %q not found", bindingLogicalCluster.Workspace.ExportName),
```

Comment:
Do we know that `.Workspace` is not `nil`?

Reply:
that's a very good question, i am not sure but i'll gate this properly 👍

Reply:
oh wait, in fact we do: https://github.com/kcp-dev/kcp/pull/1833/files/4120653e4864e6ffc04b5c3e297fec13c76bc7b2#diff-6104f7f2f6a565ab7fe344c22770310122299daf5d0f425b2a61329e88ef13a7R240

Reply:
(but to be on the safe side, this is audit, i'll gate it)

Reply:
fixed